packages/mask/src/plugins/WalletService/services/wallet/index.ts

```diff
@@ -235,20 +230,19 @@ export async function recoverWalletFromMnemonic(
     name: string,
     mnemonic: string,
     derivationPath = `${HD_PATH_WITHOUT_INDEX_ETHEREUM}/0`,
-    initialPassword?: string,
+    password?: string,
 ) {
-    const password_ = initialPassword ?? (await password.INTERNAL_getPasswordRequired())
```

Comment:
Try to get the password in-memory if the user doesn't provide it.

Reply:
No longer require password

Reply:
It's **NOT SAFTY** to restore mnemonic words and a private key w/o a password. The Mask SDK required a password for the encryption of those critical information. This means only the user who knows the password has the right to read information from the storage.

Reply:
You can update get password function as:

```ts
export async function INTERNAL_getPasswordRequired() {
    if (await hasPassword()) {
        const password_ = await INTERNAL_getPassword()
        if (!password_) throw new Error('No password set yet or expired.')
        return password_
    } else {
        return 'MASK'
    }
}
```

Reply:
There is no need for the front-end to pass down the default password.